web/src/engine/websites/RawDevart.ts

```diff
@@ -0,0 +1,98 @@
+import { Tags } from '../Tags';
+import icon from './RawDevart.webp';
+import { Chapter, DecoratableMangaScraper, Manga, type MangaPlugin, Page } from '../providers/MangaPlugin';
+import * as Common from './decorators/Common';
+import { FetchJSON, FetchRequest } from '../FetchProvider';
+
+function generateKey(): string {
+    const anonkey = String('anon_______y____').split('').map(e => '_' === e ? function () {
+        let t = '';
+        const r = 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789';
+        const n = r.length;
+        for (let o = 0; o < 1; o++) t += r.charAt(Math.floor(Math.random() * n));
+        return t;
+    }() : e).join('');
+    return anonkey;
+}
+
+const apiUrl = 'https://api.endevart.com/api/';
+
+type JSONManga = {
+     id: string, language_name: string, title: string, language: string
+}
+
+type JSONMangas = {
+    results: JSONManga[]
+}
+type JSONChapters = {
+    results: { id: string, language_name: string, chapter_number: string, cstr_full: string, language: string, name: string }[]
+}
+
+type JSONPages = {
+    pages: { id: string, image : string }[]
+}
+
+@Common.ImageDirect(true)
+
+export default class extends DecoratableMangaScraper {
+
+    public constructor() {
+        super('rawdevart', `RawDevart`, 'https://endevart.com', Tags.Language.Multilingual, Tags.Media.Manga, Tags.Media.Manhua, Tags.Media.Manhwa, Tags.Source.Aggregator);
+    }
+
+    public override get Icon() {
+        return icon;
+    }
+
+    private generateRequest(url: string, key: string): FetchRequest {
+        return new FetchRequest(url, {
+            headers: {
+                'Referer': this.URI.href,
+                'Origin': this.URI.href,
+                'X-ReadAuth': key,
+            }
+        });
+    }
+
+    public override ValidateMangaURL(url: string): boolean {
+        return /https?:\/\/endevart\.com\/comic\//.test(url);
```

Comment:
Regex can be improved as the part after `comic` consists of characters only allowed for a GUID and then must be terminated with `$`